### PR TITLE
core(fps): use observed metrics for screenshot dimensions

### PR DESCRIPTION
**Pull request by adamraine**
Closes https://github.com/GoogleChrome/lighthouse/issues/14415

The device metric dimensions only match the element bounding boxes if the page scale is 1. This PR takes the dimensions of the document element *after* the viewport is expanded to ensure the element bounding boxes line up with the screenshot.

### CODE REVIEW

File: core/gather/gatherers/full-page-screenshot.js [hunk 1]
```diff
@@ -31,8 +31,8 @@ function getObservedDeviceMetrics() {
   // Convert the Web API's kebab case (landscape-primary) to camel case (landscapePrimary).
   const screenOrientationType = kebabCaseToCamelCase(window.screen.orientation.type);
   return {
-    width: document.documentElement.clientWidth,
-    height: document.documentElement.clientHeight,
+    width: window.outerWidth,
+    height: window.outerHeight,
```
Comment (adamraine) [inline core/gather/gatherers/full-page-screenshot.js:35]: Creating the smoke test helped me discover that `document.documentElement.client*` does not represent the actual device size.

File: core/gather/gatherers/full-page-screenshot.js [hunk 2]
```diff
@@ -124,7 +131,7 @@ class FullPageScreenshot extends FRGatherer {
     const data = 'data:image/webp;base64,' + result.data;
 
     const observedDeviceMetrics =
```
[the remark this answers is not in the document]
Comment (adamraine) [reply core/gather/gatherers/full-page-screenshot.js:133]: ya good call

File: cli/test/smokehouse/test-definitions/fps-scaled.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+              left: 0,
+              right: 980,
+              width: 980,
+              height: 1742,
+            },
+          },
+          screenshot: {
+            height: 1742,
+            width: 980,
+          },
```
Comment (connorjclark) [inline cli/test/smokehouse/test-definitions/fps-scaled.js:50]: these exact value display *raw confidence* :P

hopefully we can be this exact in all environments

File: cli/test/smokehouse/test-definitions/fps-scaled.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+ * @type {Smokehouse.ExpectedRunnerResult}
+ */
+const expectations = {
+  artifacts: {
+    ViewportDimensions: {
+      innerWidth: 980,
+      // This value can vary slightly, depending on the display.
+      // https://bugs.chromium.org/p/chromium/issues/detail?id=1346355
```
Comment (connorjclark) [inline cli/test/smokehouse/test-definitions/fps-scaled.js:23]: i'm so sorry

**Comment by connorjclark** [comment]
from referenced issue

mobile:
image

(is the blurriness expected?)

mobile, but with `<meta name="viewport" content="width=device-width, initial-scale=1">`:
image

desktop:
image

cnn.com looks good.

**Comment by adamraine** [comment]
> (is the blurriness expected?)

I assume so. The actual text is tiny and then we add compression on top of that.

**Review by connorjclark — APPROVED** [review]

File: core/gather/gatherers/full-page-screenshot.js [hunk 5]
```diff
@@ -130,16 +130,16 @@ class FullPageScreenshot extends FRGatherer {
     });
     const data = 'data:image/webp;base64,' + result.data;
 
-    const documentSize =
-      await context.driver.executionContext.evaluate(getDocumentSize, {
+    const screenshotAreaSize =
+      await context.driver.executionContext.evaluate(getScreenshotAreaSize, {
```
Comment (adamraine) [inline core/gather/gatherers/full-page-screenshot.js:134]: This should usually be the same as the document size. In some cases (e.g. infinite scroll) the document element could grow larger after we expand the page size.

Comment (connorjclark) [reply core/gather/gatherers/full-page-screenshot.js:134]: This sounds like a good comment to add.

**Pull request merged**